### PR TITLE
[vpj] Consolidate controller requests for retrieving store info

### PR DESCRIPTION
**Pull request by adamxchen**
## Summary, imperative, start upper case, don't end with a period

In the VPJ, we call the controller 4 times to get the `StoreResponse` which is wasteful. For configurations, they are unlikely to change during a push job and should be cached to reduce the duplicate call. This PR caches the response so we can reduce the duplicate calls by 2.

## How was this PR tested?

GHCI

## Does this PR introduce any user-facing changes?

- [x ] No. You can skip the rest of this section.
- [] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by m-nagarajan — CHANGES_REQUESTED** [review]
Thanks @adamxchen for the cleanup.

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 1]
```diff
@@ -3214,6 +3180,42 @@ private Version getStoreVersion(String storeName, int version) {
...
+    return getStoreResponse(storeName, false);
+  }
+
+  /**
+   * Get the previously cached {@link StoreResponse} if available, otherwise query the controller.
+   * @param storeName, the store name
+   * @param refresh, if true, query the controller to get the latest store response. Usually used for up-to-date Version.
+   * @return
+   */
+  private StoreResponse getStoreResponse(String storeName, boolean refresh) {
```
[the remark this answers is not in the document]
Comment (adamxchen) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:3193]: Actually, regarding 2(b), the behavior stays as same as before. We remain same behavior in `validateStoreSettingAndPopulate` which was copied from `getSettingsFromController`.  The difference was `getStoreResponse` might refresh what `validateStoreSettingAndPopulate` changed so I've addressed this in the new revision by excluding these two fields from the method. Thanks for catching it!

[the remark this answers is not in the document]
Comment (m-nagarajan) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:3198]: Can you add a TODO as a marker that this needs further cleanup.

**Review by m-nagarajan — APPROVED** [review]
Thanks @adamxchen LGTM for now.

**Review by m-nagarajan — APPROVED** [review]

**Pull request merged**